core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/source/mysql/MysqlSource.java

```diff
@@ -0,0 +1,246 @@
+                    case 91: attributeType = AttributeType.DATE; //91 Types.DATE
+                        break;
+                    case 92: attributeType = AttributeType.DATETIME; //92 Types.TIME
+                        break;
+                    case 93: attributeType = AttributeType.DATETIME; //93 Types.TIMESTAMP
+                        break;
+                    case -2: attributeType = AttributeType.INTEGER; //-2 Types.BINARY
+                        break;
+                    case 0: attributeType = AttributeType.STRING; //0 Types.NULL
+                        break;
+                    case 1111: attributeType = AttributeType.STRING; //1111 Types.OTHER
+                        break;
+                    case 16: attributeType = AttributeType.STRING; //16 Types.BOOLEAN
+                        break;
+                    default: attributeType = AttributeType.STRING;
+                        break;
+                }
+
+                this.outputSchema = this.schemaBuilder.add(columnName, attributeType).build();
+
+            }
+            cursor = OPENED;
+        } catch (SQLException | InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new DataflowException("MysqlSink failed to connect to mysql database." + e.getMessage());
+        }
+    }
+    
+    @Override
+    public Tuple getNextTuple() throws TexeraException {
+        if (cursor == CLOSED) {
+            throw new DataflowException(ErrorMessages.OPERATOR_NOT_OPENED);
+        }
+        try {
+            if(start){
+                System.out.print(generateSqlQuery(predicate));
+                PreparedStatement ps = this.connection.prepareStatement(generateSqlQuery(predicate));
+                /*if(!predicate.getColumn().equals("") && !predicate.getKeywords().equals("")){
+                    ps.setObject(1, predicate.getColumn(), Types.VARCHAR);
+                    ps.setObject(2, predicate.getKeywords(), Types.VARCHAR);
+                    if(predicate.getLimit()!=Integer.MAX_VALUE){
+                        ps.setObject(3, predicate.getLimit(), Types.INTEGER);
+                    }
+                }
+                else{*/
+                if(predicate.getLimit()!=Integer.MAX_VALUE){
+                    ps.setObject(1, predicate.getLimit(), Types.INTEGER);
+                }
+                this.rs = ps.executeQuery();
+                start = false;}
+            //}
+			while (rs.next()) {
+			    List<IField> tb = new ArrayList();
+			    for(Attribute a: this.outputSchema.getAttributes()){
+			        if (a.getType() == AttributeType.STRING){
+			            String value = rs.getString(a.getName());
+			            value = value ==null? "":value;
+			            tb.add(new StringField(value));
+                    }else if (a.getType() == AttributeType.INTEGER){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0":value;
```

Comment:
@avinash0161 I see they are using some default values like 0 if the field is null. What do you think? I think we should also use null

Reply:
0 is misleading to the user. The only solution I see is that we have to introduce a concept of empty value. Primitive types like INTEGER are directly converted to 0 if uninitialized. So, we may have to create new 'AttributeType' which wrap over integer and have the capability of saying that the field is empty.